File: problems/easy/programmer_day/programmer_day.cpp
#include <string>
using namespace std;

/**
 * Returns the 256th day of the year in Russia for the given year.
 * @param {number} year
 * @returns {string} in format dd.mm.yyyy
 */

string dayOfProgrammer(int year) {
     if (year == 1918) {
        return "26.09." + to_string(year);
    } else if ((year >= 1700 && year <= 1917 && year % 4 == 0) ||
               (year >= 1919 && (year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)))) {
        return "12.09." + to_string(year);
    } else {
        return "13.09." + to_string(year);
    }
}

File: problems/easy/programmer_day/programmer_day.py

def dayOfProgrammer(year):
   if year == 1918:
        # Transition year: February 14 was the 32nd day
        return f"26.09.{year}"
    elif (1700 <= year <= 1917 and year % 4 == 0) or \
         (year >= 1919 and ((year % 400 == 0) or (year % 4 == 0 and year % 100 != 0))):
        # Leap year (Julian or Gregorian)
        return f"12.09.{year}"
    else:
        # Non-leap year
        return f"13.09.{year}"
